frontend/src/pages/main/topup/confirm/TopUpConfirmation.tsx

```diff
@@ -0,0 +1,108 @@
+import CardTitle from "../../../../components/typography/cardTitle/CardTitle";
+import {ICard} from "../../../../types/card";
+import {TopUpConfirmationProps} from "./props";
+import CardInformationField from "../../../../components/fields/cardIformation/CardInformationField";
+import AmountField from "../../home/merchant/components/paymentPanel/amount/AmountField";
+import {Content} from "../../../../common/constansts/content";
+import CancelButton from "../../../../components/buttons/cancelButton/CancelButton";
+import ConfirmationButton from "../../../../components/buttons/confirmationButton/ConfirmationButton";
+import {Paths} from "../../../../common/constansts/paths";
+import {useNavigate} from "react-router-dom";
+import {topUp} from "../../../../services/payment";
+import {IWalletTopUp} from "../../../../types/payment";
+import {useAppDispatch} from "../../../../stores/redux/hooks";
+import {showNotification} from "../../../../stores/redux/notification/notificationSlice";
+import {NotificationType} from "../../../../types/notification";
+import {Notifications} from "../../../../common/constansts/notifications";
+import {CardFieldType} from "../../../../types/fields";
+import "./styles.css";
+import {TopUpPlaceholder} from "../../../../common/constansts/placeholders";
+
+const initialCardData: ICard = {
+    cardName: "",
+    cardNumber: 0,
+    expirationDate: "",
+    ccv: 0
+}
+
+const initialWalletTopUp: IWalletTopUp = {
+    amount: 0
+}
+
+const TopUpConfirmation: React.FC<TopUpConfirmationProps> = ({walletId}) => {
+
+    const [cardData, setCardData] = useState<ICard>(initialCardData);
+    const [sendNotification, setSendNotification] = useState<boolean>(false);
+    const [walletTopUp, setWalletTopUp] = useState<IWalletTopUp>(initialWalletTopUp);
+    const [isAmountValid, setIsAmountValid] = useState<boolean>(false);
+    const navigate = useNavigate();
+    const dispatch = useAppDispatch();
+
+    const updateAmountChange = (amount: number) => {
+        setWalletTopUp({...walletTopUp, amount});
+    };
+
+    const updateIsAmountValidChange = (isAmountValid: boolean) => {
+        setIsAmountValid(isAmountValid);
+    };
+
+    const updateCardField = (data: string | number, cardFieldType: CardFieldType) => {
+        switch (cardFieldType) {
+            case CardFieldType.CARD_NAME:
+                setCardData({...cardData, cardName: data as string});
+                break;
+            case CardFieldType.CARD_NUMBER:
+                setCardData({...cardData, cardNumber: data as number});
+                break;
+            case CardFieldType.CARD_EXPIRATION_DATE:
+                setCardData({...cardData, expirationDate: data as string});
+                break;
+            case CardFieldType.CARD_CCV:
+                setCardData({...cardData, ccv: data as number});
+                break;
+            default:
+                break;
+        }
+
+    }
+
+    const handleNotification = () => {
```

Comment:
```suggestion
    const handleSendNotificationCheckbox = () => {
```
Otherwise, it looks like you handle actual notification